### DISCUSSION TITLE
pass not found while it is present in the /usr/bin folder

### DISCUSSION THREAD

**Discussion by az01969** [languitar/pass-git-helper#352] [Q&A] [answered]
Hi all,
I am facing a weird behavior when I am cloning git project : I got the following error:

```
_DEBUG:root:Received request "['protocol=http\n', 'host=...
DEBUG:root:Received action get with request:
{'protocol': 'http', 'host':...
DEBUG:root:Parsing mapping file. Command line: None
DEBUG:root:Parsing mapping file None
DEBUG:root:Received request "{'protocol': 'http', 'host': ...
DEBUG:root:Searching mapping to match against header 
DEBUG:root:Section "..." matches requested header "..."
DEBUG:root:Requesting entry "git" from pass_
**Unable to retrieve entry: [Errno 2] No such file or directory: 'pass'**
_fatal: could not read Username for 'http://..._
```

I checked the behavior of the failing code in the python script commands line interpreter: 
```
Python 3.8.10 (default, Nov 22 2023, 10:22:35)
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import subprocess
>>> subprocess.check_output(["pass", "show", "git"])
```
and I got the correct result without error.

Could you tell me how it happens in the pass-git-helper python code to not be able to find the pass command, please ?

Thank you for your help
Gwenn

**Comment by languitar** [comment]
Who is the caller of the git command that fails? Maybe that caller lives in an environment where `PATH` is defined differently? If in doubt, you could hack the pass-git-helper python file to include a debug statement for the PATH being used by including at the start of the file:

```python
import os
print(os.environ['PATH'])
```

**Reply by az01969** [reply]
Thank you for your debug proposal.
I could go further as the $PATH environment variable doesn't include the path to /usr/bin where pass command is.
I added the "/usr/bin" in the $PATH environment variable just before calling path, but the issue is still there:
`/bin/sh: pass: command not found
Unable to retrieve entry: Command 'pass show git' returned non-zero exit status 127.
`
It seems "pass" is not installed in this shell instance

Regards

**Reply by languitar** [reply]
Is that some shell from an IDE? Or what's so special about that shell?

**Reply by az01969** [reply]
This shell comes from the Yocto image generation

**Reply by az01969** [reply, marked as answer]
I got the confirmation pass does not work correctly with bitbake in the Yocto image generation. 
So, I use another credential git helper.

Thank you again for your support.
We can close this discussion.

Regards
Gwenn
